### DISCUSSION TITLE
work on AMD gpu (rx6800) + Ubuntu 22.04.2 LTS (Jammy Jellyfish)

### DISCUSSION THREAD

**Discussion by tornado73** [bmaltais/kohya_ss#762] [General]
install
`git clone https://github.com/bmaltais/kohya_ss.git`
`cd kohya_ss`
`python -m venv venv`
`source venv/bin/activate`
`pip install torch torchvision --index-url https://download.pytorch.org/whl/rocm5.4.2`
`pip install --use-pep517 --upgrade -r requirements.txt`
`accelerate config`

*answers*

No distributed training
no
no
no
all
fp16

`pip install tensorflow-rocm`
`sudo apt install python3-tk`
`export HSA_OVERRIDE_GFX_VERSION=10.3.0`
`source venv/bin/activate`
`python kohya_gui.py "$@"`

settings Lora of person

Caption Extension - .txt

Optimizer - AdanW

LR Scheduler - constant

Network Rank (Dimension) 128

Network Alpha 128 or 64

Enable buckets - uncheck

*Advanced Configuration*

Use xformers - uncheck

Memory efficient attention -check

Max num workers for DataLoader- 1

-------------------------------------------
start next time

open gui.sh

add

export HSA_OVERRIDE_GFX_VERSION=10.3.0
python3 kohya_gui.py "$@"

![Screenshot from 2023-05-09 22-17-56](https://github.com/bmaltais/kohya_ss/assets/58679489/37657aaf-5d4d-4b2c-a51e-c3e5aeab0034)
![Screenshot from 2023-05-09 21-40-12](https://github.com/bmaltais/kohya_ss/assets/58679489/4103d4d9-03e4-45bf-9495-22a91e8ac932)

![Screenshot from 2023-05-09 21-40-55](https://github.com/bmaltais/kohya_ss/assets/58679489/ea463a33-0428-4cb1-8507-63f75be78c45)

with other optimizers - or not work or loss=nan

Good luck

p.s. installed by me

Package                      Version
---------------------------- ----------------
absl-py                      1.4.0
accelerate                   0.18.0
addict                       2.4.0
aenum                        3.1.12
aiofiles                     23.1.0
aiohttp                      3.8.4
aiosignal                    1.3.1
altair                       4.2.2
antlr4-python3-runtime       4.9.3
anyio                        3.6.2
apturl                       0.5.2
astunparse                   1.6.3
async-timeout                4.0.2
attrs                        23.1.0
basicsr                      1.4.2
bcrypt                       3.2.0
beautifulsoup4               4.12.2
blendmodes                   2022
blinker                      1.4
boltons                      23.0.0
Brlapi                       0.8.3
cachetools                   5.3.0
certifi                      2020.6.20
chardet                      4.0.0
charset-normalizer           3.1.0
clean-fid                    0.1.29
click                        8.0.3
clip                         1.0
cmake                        3.25.0
colorama                     0.4.4
command-not-found            0.3
contourpy                    1.0.7
cryptography                 3.4.8
cupshelpers                  1.0
cycler                       0.11.0
dbus-python                  1.2.18
defer                        1.0.6
deprecation                  2.1.0
distro                       1.7.0
distro-info                  1.1build1
duplicity                    0.8.21
einops                       0.4.1
entrypoints                  0.4
facexlib                     0.3.0
fastapi                      0.94.0
fasteners                    0.14.1
ffmpy                        0.3.0
filelock                     3.9.0
filterpy                     1.4.5
flatbuffers                  23.3.3
font-roboto                  0.0.1
fonts                        0.0.3
fonttools                    4.39.3
frozenlist                   1.3.3
fsspec                       2023.4.0
ftfy                         6.1.1
future                       0.18.2
gast                         0.4.0
gdown                        4.7.1
gfpgan                       1.3.8
gitdb                        4.0.10
GitPython                    3.1.30
google-auth                  2.17.3
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
gradio                       3.28.1
gradio_client                0.2.0
grpcio                       1.54.0
h11                          0.12.0
h5py                         3.8.0
httpcore                     0.15.0
httplib2                     0.20.2
httpx                        0.24.0
huggingface-hub              0.14.1
idna                         3.3
imageio                      2.28.1
importlib-metadata           4.6.4
inflection                   0.5.1
jeepney                      0.7.1
Jinja2                       3.1.2
jsonmerge                    1.8.0
jsonschema                   4.17.3
keras                        2.11.0
keyring                      23.5.0
kiwisolver                   1.4.4
kornia                       0.6.7
language-selector            0.1
lark                         1.1.2
launchpadlib                 1.10.16
lazr.restfulclient           0.14.4
lazr.uri                     1.0.6
lazy_loader                  0.2
libclang                     16.0.0
lightning-utilities          0.8.0
linkify-it-py                2.0.2
lit                          15.0.7
llvmlite                     0.40.0
lmdb                         1.4.1
lockfile                     0.12.2
louis                        3.20.0
lpips                        0.1.4
macaroonbakery               1.3.1
Mako                         1.1.3
Markdown                     3.4.3
markdown-it-py               2.2.0
MarkupSafe                   2.1.2
matplotlib                   3.7.1
mdit-py-plugins              0.3.3
mdurl                        0.1.2
monotonic                    1.6
more-itertools               8.10.0
mpmath                       1.2.1
multidict                    6.0.4
netifaces                    0.11.0
networkx                     3.0
numba                        0.57.0
numpy                        1.22.4
oauthlib                     3.2.0
olefile                      0.46
omegaconf                    2.2.3
open-clip-torch              2.7.0
opencv-python                4.7.0.72
opt-einsum                   3.3.0
orjson                       3.8.11
packaging                    23.1
pandas                       2.0.1
paramiko                     2.9.3
pexpect                      4.8.0
piexif                       1.1.3
Pillow                       9.4.0
pip                          23.1.2
protobuf                     3.19.6
psutil                       5.9.5
ptyprocess                   0.7.0
pyasn1                       0.5.0
pyasn1-modules               0.3.0
pycairo                      1.20.1
pycups                       2.0.1
pydantic                     1.10.7
pydub                        0.25.1
PyGObject                    3.42.1
PyJWT                        2.3.0
pymacaroons                  0.13.0
PyNaCl                       1.5.0
pyparsing                    2.4.7
pyRFC3339                    1.1
pyrsistent                   0.19.3
PySocks                      1.7.1
python-apt                   2.4.0+ubuntu1
python-dateutil              2.8.2
python-debian                0.1.43ubuntu1
python-multipart             0.0.6
pytorch-lightning            1.9.4
pytorch-triton-rocm          2.0.1
pytz                         2022.1
PyWavelets                   1.4.1
pyxdg                        0.27
PyYAML                       5.4.1
realesrgan                   0.3.0
regex                        2023.5.5
reportlab                    3.6.8
requests                     2.25.1
requests-oauthlib            1.3.1
resize-right                 0.0.2
rsa                          4.9
safetensors                  0.3.1
scikit-image                 0.19.2
scipy                        1.10.1
SecretStorage                3.3.1
semantic-version             2.10.0
sentencepiece                0.1.99
setuptools                   59.6.0
six                          1.16.0
smmap                        5.0.0
sniffio                      1.3.0
soupsieve                    2.4.1
starlette                    0.26.1
sympy                        1.11.1
systemd-python               234
tb-nightly                   2.14.0a20230508
tensorboard-data-server      0.7.0
tensorboard-plugin-wit       1.8.1
tensorflow-estimator         2.11.0
tensorflow-io-gcs-filesystem 0.32.0
tensorflow-rocm              2.11.0.540
termcolor                    2.3.0
tifffile                     2023.4.12
timm                         0.6.7
tokenizers                   0.13.3
tomli                        2.0.1
toolz                        0.12.0
torch                        2.0.1+rocm5.4.2
torchaudio                   2.0.2+rocm5.4.2
torchdiffeq                  0.2.3
torchmetrics                 0.11.4
torchsde                     0.2.5
torchvision                  0.15.2+rocm5.4.2
tqdm                         4.65.0
trampoline                   0.1.2
transformers                 4.25.1
typing_extensions            4.4.0
tzdata                       2023.3
ubuntu-advantage-tools       8001
ubuntu-drivers-common        0.0.0
uc-micro-py                  1.0.2
ufw                          0.36.1
unattended-upgrades          0.1
urllib3                      1.26.5
usb-creator                  0.3.7
uvicorn                      0.22.0
wadllib                      1.3.6
wcwidth                      0.2.6
websockets                   11.0.2
Werkzeug                     2.3.3
wheel                        0.40.0
wrapt                        1.15.0
xdg                          5
xkit                         0.0.0
yapf                         0.33.0
yarl                         1.9.2
zipp                         1.0.0

**Comment by joetech** [comment]
I did get some memory issues with this still, but made a couple small adjustments and got it to run.  Should it take 30 hours (same GPU as you)?  I'm going to let it run because if the quality is good at least it'll be done, but if I do future ones, I'd love if I could get a quality LORA without taking 30 hours. It dropped to about 9 hours when I moved from bf16 to fp16 (now ~2.5s/it)

**Comment by hqnicolas** [comment]
@joetech  @tornado73  Today I turn my Gist into an ROCm Docker REPO.
Please if a Mantainer can Analyse and ADD this AMD Docker Compose to the **bmaltais/kohya_ss** REPO
https://github.com/hqnicolas/bmaltaisKohya_ssROCm
With this files you can Run Kohya_ss into AMD RDNA Consumer Graphics Series with Docker rocm/pytorch:rocm6.1
if You like it don't forget to leave a star on my repo ⭐

**Comment by Insistentelk** [comment]
Unfortunately  at the current moment (9-24) this no longer works as the -r requirements.txt(line 35) breaks due to not bring a python project.
